### PR TITLE
Support for mocha's 'qunit' interface

### PR DESCRIPTION
**Pull request by Flaise**

### CODE REVIEW

File: lib/frameworks/mocha.js [hunk 1]
```diff
@@ -18,7 +19,7 @@ module.exports.run = function(config, specs, capabilities) {
         requires = config.mochaOpts.require,
         runner;
 
-    if(typeof config.mochaOpts.ui !== 'string' || !config.mochaOpts.ui.match(/(bdd|tdd)/i)) {
+    if(typeof config.mochaOpts.ui !== 'string') {
```
Comment (Flaise) [inline lib/frameworks/mocha.js]: Mocha may have new interfaces added to it in the future. This sort of thing shouldn't be hard-coded.

File: lib/frameworks/mocha.js [hunk 2]
```diff
@@ -46,7 +47,7 @@ module.exports.run = function(config, specs, capabilities) {
     /**
      * enable generators if supported
      */
-    if(hasES6Support) {
+    if(hasES6Support && interfaces[config.mochaOpts.ui]) {
```
Comment (Flaise) [inline lib/frameworks/mocha.js]: This prevents the above changes from breaking anything for interfaces you don't support.

File: lib/frameworks/mocha.js (hunk 1 above)
Comment (christian-bromann) [inline lib/frameworks/mocha.js]: if we have it hard coded in the `interfaces` object why not here?

File: lib/frameworks/mocha.js (hunk 2 above)
Comment (christian-bromann) [inline lib/frameworks/mocha.js]: why did you add qunit in the interface at all?

File: lib/frameworks/mocha.js (hunk 1 above)
Comment (Flaise) [inline lib/frameworks/mocha.js]: Because writing it this way makes it so that the worst that happens for interfaces you don't support is that generators won't work. The worst case for the way you wrote it is that it reverts to an entirely different interface.

File: lib/frameworks/mocha.js (hunk 2 above)
Comment (Flaise) [inline lib/frameworks/mocha.js]: Mocha has 5 interfaces. https://mochajs.org/#interfaces One of my projects uses one that you (without this PR) don't support.

**Comment by christian-bromann** [comment]
Ok, looks good. Thanks for the explanations.
